ultravox/tools/eval_tool.py

```diff
@@ -1,35 +1,78 @@
-import argparse
+#!/usr/bin/env python
 import dataclasses
-from typing import IO
+import logging
+import os
+import sys
 
 import simple_parsing
+import torch
+import torch.distributed as dist
+import wandb
 
 from ultravox.evaluation import eval
-from ultravox.evaluation import eval_types
+from ultravox.inference import ultravox_infer
+from ultravox.training import config_base
+from ultravox.training import ddp_utils
+from ultravox.utils import device_helpers
+from ultravox.utils import string_helpers
 
+logging.basicConfig(level=logging.INFO)
 
-@dataclasses.dataclass
-class EvalArgs:
-    # Path to the audio file
-    file: IO = simple_parsing.field(type=argparse.FileType("rb"), alias="-f")
-    # Metric to use for evaluation (e.g., "asr" or "boolq")
-    metric: str = simple_parsing.field(default="asr", alias="-m")
-    # Verbose output
-    verbose: bool = simple_parsing.field(default=False, alias="-v")
 
+def main():
+    args = simple_parsing.parse(
+        config_class=config_base.TrainConfig,
+        add_config_path_arg=True,
+        args=[string_helpers.fix_hyphens(arg) for arg in sys.argv[1:]],
+    )
 
-def main(args: EvalArgs):
-    scores = []
-    for i, line in enumerate(args.file.readlines()):
-        sample = eval_types.Sample.from_json(line)
-        result = eval.evaluate_answer(sample, metric=args.metric)
-        assert result.score is not None, "Rating failed."
-        scores.append(result.score)
-        average = sum(scores) / len(scores)
-        print(f"{i}: score={result.score:.2f} average={average:.2f}")
-        if args.verbose and isinstance(result, eval_types.InstructResult):
-            print(f"  reason={result.reason}")
+    world_size = device_helpers.get_world_size()
+    local_rank = device_helpers.get_local_rank()
+    device = torch.device(args.device, index=local_rank)
+    if world_size > 1:
+        dist.init_process_group(backend="gloo")
+
+    if local_rank == 0:
+        args.output_dir.mkdir(parents=True, exist_ok=True)
+        if "wandb" in args.report_logs_to:
+            wandb.init(
+                project=os.getenv("WANDB_PROJECT", "ultravox"),
+                config=dataclasses.asdict(args),
+                name=args.exp_name,
+                dir="runs",
+            )
```

Comment:
Nit: A suggestion is to add the option to set the `id` field so that the metrics could be reported on the original W&B run.